repos/system_upgrade/common/libraries/repofileutils.py

```diff
@@ -11,6 +11,15 @@
     api.current_logger().warning('repofileutils.py: failed to import dnf')
 
 
+class InvalidRepoDefinition(Exception):
+    def __init__(self, msg, repofile, repoid):
```

Comment:
Since this is public and in a shared library, add a docstring describing what is the exception used for.

Also update the docstring in the `parse_repofile` function to document that it can raise this exception and when.

Reply:
Tests were added and docstrings were updated.